### ISSUE TITLE
Missing storage_offset support prevents full on-Spyre paged attention

### ISSUE THREAD

**Issue by jvlunteren** [torch-spyre/torch-spyre#2906]
### What happened?

The spyre-inference paged‑KV attention backend (`spyre_attn.py`) cannot run fully on-device because device-side ops reading from sliced tensors (`storage_offset > 0`) ignore the offset and instead read from the start of the underlying storage.

This affects all stages of the paged-attention pipeline, including KV cache updates and reads, and the online softmax loop, since they rely heavily on views into larger tensors. As a result, any operation that consumes a sliced device tensor silently reads incorrect data.

The failure is silent at the PyTorch level: tensors report correct metadata (`is_contiguous`, `storage_offset`, `data_ptr`), and ops complete successfully but produce values corresponding to offset 0 rather than the intended slice.

### What did you expect to happen?

Sliced tensors should be read correctly and produce the expected results.

### How can we reproduce it?

```
import torch
import torch_spyre

DEVICE = "spyre"

x = torch.tensor(
    [[1., 1., 1., 1.],
     [2., 2., 2., 2.],
     [3., 3., 3., 3.],
     [4., 4., 4., 4.]],
    dtype=torch.float16,
).to(DEVICE)

view = x[2]                  # row 2 = [3, 3, 3, 3], storage_offset=8

print("plain D2H :", view.cpu().tolist())          # [3, 3, 3, 3]  PASS
print("clone()   :", view.clone().cpu().tolist())  # [1, 1, 1, 1]  FAIL
print("view + 0  :", (view + 0).cpu().tolist())    # [1, 1, 1, 1]  FAIL
print("view * 1  :", (view * 1).cpu().tolist())    # [1, 1, 1, 1]  FAIL

buf = torch.empty_like(view)
buf.copy_(view)
print("empty+copy:", buf.cpu().tolist())           # [1, 1, 1, 1]  FAIL

# Workaround: round-trip through CPU
fixed = view.cpu().to(DEVICE)
print("round-trip:", fixed.cpu().tolist())         # [3, 3, 3, 3]  PASS
```

### Any environmental details we need to know?

- torch-spyre: a6baa9c73b2de0f1b199c2c567d8609720bcf5a8 (2026‑06‑26)
- spyre-inference: v0.1.dev95 (editable install)
- torch: 2.11.0
- dtype: fp16
- single Spyre device

### Anything else we need to know?

Current workarounds in `spyre_attn.py` avoid sliced device reads using CPU-fallbacks (introducing per-token round-trips in the hot path). These workarounds are functional but come with significant cost: they prevent a fully on-device implementation of paged attention, limit opportunities for fusion, and add unnecessary data movement. Fixing correct read behavior for sliced tensors would allow the backend to operate entirely on-device as intended.

Multiple issues (#1333, #894, #1548, #1549, #2426, #2561, #2816) cover parts of the `storage_offset` problem, but none address it from the paged-attention perspective, where it blocks scatter, gather, and online softmax. The ongoing indirect-addressing work (#2469, #2699, #2725, #2433)  could avoid the need for `storage_offset` in scatter/gather by computing addresses at runtime instead of using sliced views. In that sense, it overlaps with this use case. However, the storage_offset issue is broader. It also affects online-softmax intermediates that are views into larger tensors, which indirect addressing does not cover. So they are complementary.

### Relevant log output

```Shell

```

**Comment by jvlunteren** [comment]
This issue is particularly relevant for implementations in which the tile size (i.e., the granularity at which K and V are processed) differs from the page size (similar to our Triton attention kernels in vLLM). In such cases, slicing is probably necessary to achieve good performance, while also substantially reducing the number of cached graphs.

I will close this issue for now, as we are currently focusing on an implementation with matching tile and page sizes, where this concern is less relevant. I can reopen or resubmit it if it becomes relevant again.
